Frameworks/UIKit/UIApplicationMain.mm

```diff
@@ -384,8 +394,27 @@ void _UIApplicationShutdown() {
 }
 
 extern "C" void UIApplicationMainHandleProtocolEvent(IInspectable* protocolUri, const wchar_t* sourceApplication) {
-    WFUri* protocolResult = [WFUri createWith:protocolUri];
-    NSString* source = _bundleIdFromPackageFamilyName(sourceApplication);
+    id result = _createProjectionObject("WFUri", protocolUri, "protocol activation");
+
+    if (result) {
+        NSString* source = _bundleIdFromPackageFamilyName(sourceApplication);
+
+        [[UIApplication sharedApplication] _sendProtocolReceivedEvent:result source:source];
+    }
+}
+
+id _createProjectionObject(const char* className, IInspectable* source, const char* description) {
```

Comment:
>_createProjectionObject [](start = 3, length = 23)

should we switch to this approach for UIView's public xamlElement property, and for initWith:xamlElement?